testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -1406,43 +1405,57 @@ public void testUnsupportedColumnMappingModeChangeDataFeed(String mode)
-                    .hasMessageContaining("Unsupported column mapping mode for tables with change data feed enabled: " + mode.toUpperCase(ENGLISH));
-            assertQueryFailure(() -> onTrino().executeQuery("DELETE FROM delta.default." + targetTableName))
-                    .hasMessageContaining("Unsupported column mapping mode for tables with change data feed enabled: " + mode.toUpperCase(ENGLISH));
-            assertQueryFailure(() -> onTrino().executeQuery("MERGE INTO delta.default." + targetTableName + " cdf USING delta.default." + sourceTableName + " n " +
+            onTrino().executeQuery("INSERT INTO default." + targetTableName + " VALUES (3, 'nation3', 300)");
+
+            // Column mapping mode 'none' is tested in TestDeltaLakeChangeDataFeedCompatibility
+            onTrino().executeQuery("UPDATE delta.default." + targetTableName + " SET regionkey = 10 WHERE regionkey = 100");
+            onTrino().executeQuery("DELETE FROM delta.default." + targetTableName + " WHERE regionkey = 200");
+            onTrino().executeQuery("MERGE INTO delta.default." + targetTableName + " cdf USING delta.default." + sourceTableName + " n " +
                     "ON (cdf.nationkey = n.nationkey) " +
                     "WHEN MATCHED " +
                     "THEN UPDATE SET nationkey = (cdf.nationkey + n.nationkey + n.regionkey) " +
                     "WHEN NOT MATCHED " +
-                    "THEN INSERT (nationkey, name, regionkey) VALUES (n.nationkey, n.name, n.regionkey)"))
-                    .hasMessageContaining("Unsupported column mapping mode for tables with change data feed enabled: " + mode.toUpperCase(ENGLISH));
+                    "THEN INSERT (nationkey, name, regionkey) VALUES (n.nationkey, n.name, n.regionkey)");
 
             assertThat(onDelta().executeQuery("SELECT nationkey, name, regionkey, _change_type, _commit_version " +
                     "FROM table_changes('default." + targetTableName + "', 0)"))
                     .containsOnly(
                             row(1, "nation1", 100, "insert", 1),
                             row(2, "nation2", 200, "insert", 2),
-                            row(3, "nation3", 300, "insert", 3));
+                            row(3, "nation3", 300, "insert", 3),
+                            row(1, "nation1", 100, "update_preimage", 4),
+                            row(1, "nation1", 10, "update_postimage", 4),
+                            row(2, "nation2", 200, "delete", 5),
+                            row(4, "nation4000", 4000, "insert", 6),
+                            row(1002, "nation1", 10, "update_postimage", 6),
+                            row(1, "nation1", 10, "update_preimage", 6));
         }
         finally {
             dropDeltaTableWithRetry("default." + targetTableName);
         }
     }
 
+    @DataProvider
+    public Object[][] mappingModesProvider()
```

Comment:
FYI, columnMappingDataProvider already exists.

Reply:
omg I didn't even use this one :|